src/services/tx/safeUpdateParams.ts

```diff
@@ -1,11 +1,41 @@
 import { MetaTransactionData, OperationType } from '@gnosis.pm/safe-core-sdk-types'
 import { ChainInfo, SafeInfo } from '@gnosis.pm/safe-react-gateway-sdk'
-import { getFallbackHandlerContractInstance, getGnosisSafeContractInstance } from '@/services/contracts/safeContracts'
+import {
+  getFallbackHandlerContractInstance,
+  getGnosisSafeContractInstance,
+  _getSafeContractDeployment,
+} from '@/services/contracts/safeContracts'
 import { LATEST_SAFE_VERSION } from '@/config/constants'
+import { isValidSafeVersion } from '@/hooks/coreSDK/safeCoreSDK'
+import { Gnosis_safe__factory } from '@/types/contracts/factories/Gnosis_safe__factory'
+import { getWeb3 } from '@/hooks/wallets/web3'
 
 export const CHANGE_MASTER_COPY_ABI = 'function changeMasterCopy(address _masterCopy)'
 export const CHANGE_FALLBACK_HANDLER_ABI = 'function setFallbackHandler(address handler)'
 
+const getContractInterface = (chain: ChainInfo, version: string) => {
+  if (isValidSafeVersion(version)) {
```

Comment:
This check happens in `getSafeContractInstance` itself. I considered using `try { ... } catch { ... }` instead but thought this was more declarative.